action.yaml

```diff
@@ -72,17 +82,43 @@ runs:
           PRE_COMMIT_ARGS="--files ${{ steps.changed-files.outputs.all_changed_files }}"
           echo "PRE_COMMIT_ARGS=$PRE_COMMIT_ARGS" >> "$GITHUB_ENV"
         fi
+
+    - name: Compute pre-commit cache key
+      if: inputs.s3-bucket-name != ''
+      id: pre_commit_cache_key
+      run: |
+        echo "key=${{ hashFiles('**/.pre-commit-config.yaml') }}" >> $GITHUB_OUTPUT
```

Comment:
Nice! But so if we make a change to cspell for example, which we typically install as an `additional_dependencies` as a floating major, if the hooks are cached, when we ship a new minor of the dictionary, will we still be stuck with the cached version?

Reply:
I'm not sure of this and we would have to test it out. The pre-commit docs don't mention anything extra on how to cache things https://pre-commit.com/#managing-ci-caches

Reply:
I think we'll have to test or think about using a pinned version (if cspell doesn't change that often it might be ok) and renovatebot might pick it?

Reply:
As discussed offline we can test what happens and worst case we can update cspell config to reference the full version

Reply:
And tweak renovatebot so it can detect this dependency and update it accordingly